src/main/kotlin/blackjack/domain/Player.kt

```diff
@@ -0,0 +1,43 @@
+package blackjack.domain
+
+class Player(val name: String, private val deck: Deck) {
+    val cards = mutableListOf<Card>()
+
+    init {
+        repeat(DEFAULT_CARD_COUNT) { addCard(deck.draw()) }
+    }
+
+    fun play(
+        isDrawCard: (String) -> Boolean,
+        onDrawCard: () -> Unit,
+        onExitPlay: () -> Unit,
+    ) {
+        while (isDrawCard(name)) {
+            val card = deck.draw()
+            if (canAdd(card)) {
+                addCard(card)
+                onDrawCard()
+            } else {
+                break
+            }
+        }
+        onExitPlay()
+    }
+
+    fun getCardSum(): Int {
+        return cards.sumValues()
+    }
```

Comment:
상태값을 구한다면 읽기전용 프로퍼티를 이용하여 표현해보는것도 좋을것 같습니다.

```suggestion
    val cardSum: Int 
        get() = cards.sumValues()
```

Reply:
감사합니다.